python/taichi/lang/ast/ast_transformer.py

```diff
@@ -772,6 +772,17 @@ def build_Attribute(ctx, node):
             index = build_stmt(ctx, node.value.slice)
             node.value.ptr = None
             node.ptr = lambda val: append(x.parent(), index, val)
+        elif node.attr == "deactivate" and isinstance(node.value,
```

Comment:
The logic processing `append` and `deactivate` are very similar, so maybe we can merge them together?